### ISSUE TITLE
Unlinked references does not work with evil

### ISSUE THREAD

**Issue by oatmealm** [NicolasPetton/nroam#11]
Wonderful. Really like the way it's implemented. Another advantage is of course that working with a sidebar is not ideal when the frame need to be kept very narrow (i.e. alongside another app).

How does the unlinked feature work, BTW? Right now, with evil enabled (in Doom Emacs) when in normal mode nothing happens (I'm assuming there are many unlinked references) when I push or click with the mouse. When I switch to "insert mode" push the org button (press), the text breaks (new line is inserted).

**Comment by Whil-** [comment]
@oatmealm what operating system are you using?

Personally I'm on windows and have not gotten this to work either. I've traced it to the unlinked references functionality in Org roam not working currently. Not unlikely that this problem is due to my own config though. Haven't dug further.

**Comment by oatmealm** [comment]
> @oatmealm what operating system are you using?

I'm on Fedora Linux, Emacs 27.1.

How is it used in plain org-roam?

**Comment by NicolasPetton** [comment]
@oatmealm @Whil- you need ripgreg for `org-roam-unlinked-references` to work:

> Requires a version of Ripgrep with PCRE2 support installed, with the executable 'rg' in variable exec-path.

**Comment by NicolasPetton** [comment]
> How is it used in plain org-roam?

In plain org-roam you call `org-roam-unlinked-references`.

**Comment by Whil-** [comment]
> @oatmealm @Whil- you need ripgreg for `org-roam-unlinked-references` to work:
> 
> > Requires a version of Ripgrep with PCRE2 support installed, with the executable 'rg' in variable exec-path.

For me there is something else as well. Because rg with PCRE2 is installed. But there is something wrong there with how it integrate into the windows shell. Because running the exact same commands that Org roam produces in Powershell works, but not in cmd.exe. Oh, well. It's not critical for me and surely not nroam that is the culprit anyhow. :)

**Comment by NicolasPetton** [comment]
@Whil- are you sure it's in your `exec-path`?

**Comment by NicolasPetton** [comment]
> It's not critical for me and surely not nroam that is the culprit anyhow. :)

Indeed, I'm closing this one :-)

**Comment by Whil-** [comment]
> @Whil- are you sure it's in your `exec-path`?

Yes, I'm using rg.el quite heavily :) (Which is also the reason for not really _needing_ the unlinked references ;)

**Comment by oatmealm** [comment]
@NicolasPetton it is installed and in the path... (find-executable "rg") can find it.
I'm not sure though about PCRE2 ... how can I check?

Also, invoking `org-roam-unlinked-references` seems to work. A buffer is created and populated with results.

```
rg ---version
ripgrep 12.1.1
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)
```

**Comment by NicolasPetton** [comment]
Ok, reopening then.

**Comment by NicolasPetton** [comment]
@oatmealm could you paste the stacktrace when you click on the link to show unlinked references in nroam?

**Comment by oatmealm** [comment]
There's no error, so no stacktrace to provide. I've enabled `toggle-debug-on-error`.

I think there's "No further link found" echoed, but I'm not sure if it's for linked or unlinked....

Sorry about the mixed scripts, but it is in fact for the same buffer.

With `nroam`

![gnome-shell-screenshot-PJB4Z0](https://user-images.githubusercontent.com/68159077/110955250-b0621000-8349-11eb-8a4a-64708f749499.png)


With `org-roam-unlinked-refrences`

![gnome-shell-screenshot-2TUXZ0](https://user-images.githubusercontent.com/68159077/110955287-b9eb7800-8349-11eb-91f7-6845854f5b48.png)

**Comment by NicolasPetton** [comment]
@oatmealm what happens when you press `RET` on the "[Show unlinked references]" button? Could you paste a screenshot?

**Comment by oatmealm** [comment]
Attached.

![gnome-shell-screenshot-7DBSZ0](https://user-images.githubusercontent.com/68159077/110956676-274bd880-834b-11eb-9d9d-21af90db776c.png)


Ok. I think I see what's happening.

As I mentioned, `evil-mode` is active. When in `normal` mode, nothing happens when I press or click the "button". When I switch the buffer to `insert` mode, the button does not seem to be set to read-only. When I switch to emacs state and click, it works.

**Comment by oatmealm** [comment]
PS. Amazing that it looks up references

**Comment by NicolasPetton** [comment]
@oatmealm just tried with `evil` and it works for me in normal mode, so I cannot reproduce the issue locally.

**Comment by oatmealm** [comment]
How could I trace this in my setup? What should I be looking for?

**Comment by NicolasPetton** [comment]
@oatmealm what do you get if you move the point somewhere within the button and do `C-h k RET`? I get `push-button`.

**Comment by oatmealm** [comment]
@NicolasPetton Seems like Doom overrides the behavior... guess I need to define an nroam button type somehow?

```
Signature
(+org/dwim-at-point &optional ARG)

Documentation
Do-what-I-mean at point.

If on a:
- checkbox list item or todo heading: toggle it.
- clock: update its time.
- headline: cycle ARCHIVE subtrees, toggle latex fragments and inline images in
  subtree; update statistics cookies/checkboxes and ToCs.
- footnote reference: jump to the footnote's definition
- footnote definition: jump to the first reference of this footnote
- table-row or a TBLFM: recalculate the table's formulas
- table-cell: clear it and go into insert mode. If this is a formula cell,
  recaluclate it instead.
- babel-call: execute the source block
- statistics-cookie: update it.
- latex fragment: toggle it.
- link: follow it
- otherwise, refresh all inline images in current tree.
```

**Comment by NicolasPetton** [comment]
@oatmealm I'm not sure there's a way to fix that that within nroam, as Doom is overriding the default behavior of buttons.
You could try to advice `+org/dwim-at-point`, it should fix the problem but it would be quite hackish.

**Comment by NicolasPetton** [comment]
@oatmealm @Whil- could you check if commit https://github.com/NicolasPetton/nroam/commit/cab2be0102c122a8ef4607a1a12708bcb05b5afd fixes the issue?

**Comment by Whil-** [comment]
> @Whil- could you check if commit cab2be0 fixes the issue?

No difference on my end. But I'm fairly certain my issue lies somewhere else. Either Org roam itself (most likely) or my own config (ehrm... also likely ofc... but I think I've isolated this from my config before).

**Comment by oatmealm** [comment]
In Doom the "show unlinked references" is active when I activate "emacs state" temporarly.... the las update didn't solve the problem.

**Comment by NicolasPetton** [comment]
Ok, I think I'll have to setup Doom then to reproduce the issue.

**Comment by mediapathic** [comment]
Have the same issue here, also on doom. Can confirm (after #cab2be0 ) that `RET` acts as `+org/dwim-at-point`, but if I call `push-button` while point is on the button it works fine (also holy crap this is a great feature I hadn't noticed before).

**Comment by mediapathic** [comment]
I talked with Henrik (the doom maintainer) about this, and we worked out that `+org/dwim-at-point` should probably push buttons as well. He's going to put an update for that in a future doom version, so @NicolasPetton probably ok to hold off work on this one for now. In the meantime he gave me this to put in my config, which fixes the problem for now: 

```
(defadvice! do-the-thing (orig-fn &rest args)
  :around #'+org/dwim-at-point
  (if (button-at (point))
      (call-interactively #'push-button)
    (apply orig-fn args)))
```

You probably want to call it something other than do-the-thing.
